src/trio/_core/_run.py

```diff
@@ -550,6 +555,13 @@ def __enter__(self) -> Self:
                 "Each CancelScope may only be used for a single 'with' block"
             )
         self._has_been_entered = True
+
+        if self._relative_deadline is not None:
+            if self._relative_deadline + current_time() < self._deadline:
+                self._deadline = self._relative_deadline + current_time()
+            else:
+                self._relative_deadline = self.deadline - current_time()
```

Comment:
We don't need to update `_relative_deadline` I think.

Reply:
ah yeah, that's a relic of previous version.